protos/sources/vega/governance.proto

```diff
@@ -352,6 +352,76 @@ message ProposalTerms {
+  // constrained by `minEnact` and `maxEnact` network parameters.
+  int64 enactment_timestamp = 1;
+  // Changes being proposed.
+  oneof change {
+    // Proposal change for modifying an existing futures market.
+    UpdateMarket update_market = 101;
+    // Proposal change for creating new futures market.
+    NewMarket new_market = 102;
+    // Proposal change for updating Vega network parameters.
+    UpdateNetworkParameter update_network_parameter = 103;
+    // Proposal change for a freeform request, which can be voted on but does not change the behaviour of the system,
+    // and can be used to gauge community sentiment.
+    NewFreeform new_freeform = 104;
+    // Proposal change for updating an asset.
+    UpdateAsset update_asset = 105;
+    // Proposal change for creating new spot market.
+    NewSpotMarket new_spot_market = 106;
+    // Proposal change for modifying an existing spot market.
+    UpdateSpotMarket update_spot_market = 107;
+    // Proposal change for a governance initiated transfer.
+    NewTransfer new_transfer = 108;
+    // Proposal change to cancel a governance initiated transfe.
+    CancelTransfer cancel_transfer = 109;
+    // Proposal change for updating the state of a market.
+    UpdateMarketState update_market_state = 110;
+    // Proposal change for updating the referral program.
+    UpdateReferralProgram update_referral_program = 111;
+    // Proposal change for updating the volume discount program.
+    UpdateVolumeDiscountProgram update_volume_discount_program = 112;
+  }
+}
+
+message ProposalParameters {
+  // Represents the minimum time before a proposal can be closed for voting.
+  int64 min_close = 1;
+  // Represents the maximum time that a proposal can be open for voting.
+  int64 max_close = 2;
+  // Represents the minimum time before an enacted proposal takes effect.
+  int64 min_enact = 3;
+  // Represents the maximum time before an enacted proposal must take effect.
+  int64 max_enact = 4;
+  // Specifies the required percentage of eligible participants that must partake in the proposal for it to be valid.
+  string required_participation = 5;
+  // Specifies the required percentage of votes a proposal needs to be approved.
+  string required_majority = 6;
+  // Specifies the minimum governance token balance a proposer must hold to initiate a proposal.
+  string min_proposer_balance = 7;
+  // Specifies the minimum governance token balance a voter must hold to participate in voting.
+  string min_voter_balance = 8;
+  // Specifies the required percentage of participation from liquidity providers for the proposal to be valid.
+  string required_participation_lp = 9;
+  // Specifies the required majority percentage from liquidity providers for a proposal to be approved.
+  string required_majority_lp = 10;
+  // Specifies the minimum share that is akin to equity, which a participant must hold for their vote to be eligible.
+  string min_equity_like_share = 11;
+}
```

Comment:
Do we need to add this? Assuming that the most restrictive parameters are to be used accross the batch, not sure we need to add these things here?

Reply:
@jeremyletang well because of snapshot. But give what @davidsiska-vega said I have not implemented this correctly. So will need to remove it.

Reply:
@davidsiska-vega true, my point is that, as all proposal needs to pass, the only parameters relevant are the most restrictives accross all proposals somehow.

Reply:
@jeremyletang You seem to be right. We still need to evaluate each proposal individually because of the LP votes. But we only need to store the most restrictive numbers on participation and majority for token holders as they all need to pass.